### PR TITLE
[ops, model, config] feat: support Qwen3-Omni-MoE training on Ascend NPU with fused RoPE

### PR DESCRIPTION
**Pull request by vvyuervv**
### What does this PR do?

> This PR introduces training support and hardware-specific optimizations for the **Qwen3-Omni-MoE** model on **Ascend NPU** architectures within VeOmni.

> To maintain a clean codebase and prevent regression or coupling with existing GPU paths, **all NPU-specific training and modeling logic has been extracted into standalone files**. 

> The core optimization in this PR replaces the network's native eager RoPE implementation with the specialized **`torch_npu.npu_rotary_mul` fused operator**. The original native RoPE path generates a massive volume of separate element-wise `mul` and `add` computations during the rotary embedding phase. By fusing these redundant element-wise operations into a single kernel via `npu_rotary_mul`, we eliminate the intermediate overhead of individual calculation steps, significantly boosting RoPE compute performance.

### Checklist Before Starting

- Search for relative PRs/issues and link here: ...
- PR title follows `[{modules}] {type}: {description}` format (enforced by [check_pr_title.yml](.github/workflows/check_pr_title.yml))
  - **Allowed modules:** `agent`, `ci`, `ckpt`, `config`, `data`, `dist`, `docker`, `docs`, `logging`, `lora`, `misc`, `model`, `omni`, `optim`, `ops`, `parallel`, `perf`, `release`, `task`, `trainer`
  - **Allowed types:** `feat`, `fix`, `refactor`, `chore`, `test`
  - Breaking changes: prepend `[BREAKING]` — e.g. `[BREAKING][parallel, model] feat: dynamic batching`

### Test

> Verification was conducted on the Ascend 910B, and the qwen3-omni-moe model can be trained. In ROPE, replacing certain fusion operators with others improved performance by more than 3 times.

q_embed = (q * cos) + (rotate_half(q) * sin)     --> q_embed =  npu_rotary_mul(q, cos, sin, rotary_mode='half')

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/Z00443~1/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/Z00443~1/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">

Type | Duration(us) | Type | Duration(us) | 加速比
-- | -- | -- | -- | --
Mul | 464.739 | RotaryPositionEmbedding | 1121.565 | 3.026642
Slice | 509.96
Neg | 220.549
Slice | 453.018
ConcatD | 490.8
Mul | 424.137
Add | 831.373

</body>

</html>

> We set identical random seeds and input noise, then trained VeOmni-Qwen3-Omni-MoE with two distinct RoPE implementations: one leveraging the fused torch_npu.npu_rotary_mul operator, and the other using eager-mode RoPE. After 306 training steps, the discrepancies in total loss and gradient norm between the two training runs both equal zero, which verifies that these two RoPE implementations are mathematically equivalent.

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/Z00443~1/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/Z00443~1/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<div data-component=markdown-code style='box-sizing: border-box;-webkit-font-smoothing: antialiased;
text-rendering: geometricprecision;font-variant-ligatures: normal;font-variant-caps: normal;
orphans: 2;text-align:start;widows: 2;-webkit-text-stroke-width: 0px;
text-decoration-thickness: initial;text-decoration-style: initial;text-decoration-color: initial'><span
style='box-sizing: border-box;-webkit-font-smoothing: antialiased;text-rendering: geometricprecision'><span
style='box-sizing: border-box;-webkit-font-smoothing: antialiased;text-rendering: geometricprecision'></div>

</span></span>

<body link=blue vlink=purple>

step | eager_total_loss | fused_total_loss | Comparison absolute | eager_grad_norm | fused_grad_norm | Comparison absolute
-- | -- | -- | -- | -- | -- | --
0/306 | 9.3983 | 9.3983 | 0 | 46.7743 | 46.7743 | 0
1/306 | 8.6941 | 8.6941 | 0 | 44.9998 | 44.9998 | 0
2/306 | 7.0659 | 7.0659 | 0 | 37.0172 | 37.0172 | 0
3/306 | 6.5453 | 6.5453 | 0 | 32.5434 | 32.5434 | 0
4/306 | 6.2069 | 6.2069 | 0 | 44.2188 | 44.2188 | 0
… | … | … | … | … | … | …
150/306 | 1.3464 | 1.3464 | 0 | 0.8647 | 0.8647 | 0
151/306 | 1.3331 | 1.3331 | 0 | 0.808 | 0.808 | 0
152/306 | 1.3622 | 1.3622 | 0 | 0.8288 | 0.8288 | 0
153/306 | 1.3162 | 1.3162 | 0 | 0.8648 | 0.8648 | 0
154/306 | 1.3427 | 1.3427 | 0 | 0.8392 | 0.8392 | 0
… | … | … | … | … | … | …
300/306 | 1.2305 | 1.2305 | 0 | 0.7569 | 0.7569 | 0
301/306 | 1.2277 | 1.2277 | 0 | 0.7753 | 0.7753 | 0
302/306 | 1.2354 | 1.2354 | 0 | 0.8001 | 0.8001 | 0
303/306 | 1.219 | 1.219 | 0 | 0.8085 | 0.8085 | 0
304/306 | 1.2312 | 1.2312 | 0 | 0.7795 | 0.7795 | 0

</body>

</html>

### API and Usage Example

> Show API changes and usage examples if applicable.

### Design & Code Changes

> High-level design description and specific change list.

### Checklist Before Submitting

- Read the [Contribute Guide](https://github.com/ByteDance-Seed/VeOmni/blob/main/CONTRIBUTING.md)
- Applied pre-commit checks
- Added/updated documentation
- If `tasks/` training scripts were moved or renamed: updated `docs/` examples and verified `python3 scripts/ci/check_doc_task_paths.py` passes (also enforced by the **Check doc task paths** CI workflow)
- Added tests to CI workflow (or explained why not feasible)

### CODE REVIEW

**Comment by Crystal-jiang** [comment]
Please add precision alignment results (before vs. after the change) for around 10 steps.

File: veomni/models/transformers/qwen3_omni_moe/qwen3_omni_moe_npu_patch_gen_config.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+def apply_rotary_pos_emb_version(
+    q: torch.Tensor, k: torch.Tensor, cos: torch.Tensor, sin: torch.Tensor
+) -> tuple[torch.Tensor, torch.Tensor]:
+    cos = cos.unsqueeze(-2).float()
+    sin = sin.unsqueeze(-2).float()
+
+    q_embed = npu_rotary_mul(q.float(), cos, sin, rotary_mode="half").to(q.dtype)
+    k_embed = npu_rotary_mul(k.float(), cos, sin, rotary_mode="half").to(k.dtype)
+
+    return q_embed, k_embed
```
[the remark this answers is not in the document]
Comment (vvyuervv) [reply veomni/models/transformers/qwen3_omni_moe/qwen3_omni_moe_npu_patch_gen_config.py]: apply_rotary_pos_emb_version has been modified to apply_rotary_pos_emb_vision.

**Review by Coach257 — APPROVED** [review]
